utils/run-unasync.py

```diff
@@ -16,42 +16,97 @@
+        for pattern in patterns:
+            subprocess.check_call(["sed", "-i.bak", pattern, str(path)])
+        subprocess.check_call(["rm", f"{path}.bak"])
+
+
+def format_dir(dir: Path):
+    subprocess.check_call(["isort", "--profile=black", dir])
+    subprocess.check_call(["black", dir])
+
+
+def run(
+    rule: unasync.Rule,
+    cleanup_patterns: list[str] = [],
+    format: bool = False,
+):
+    root = Path(__file__).absolute().parent.parent
+    source_dir = root / rule.fromdir.lstrip("/")
+    output_dir = root / rule.todir.lstrip("/")
 
     filepaths = []
-    for root, _, filenames in os.walk(
-        Path(__file__).absolute().parent.parent / "elasticsearch/_async"
-    ):
+    for root, _, filenames in os.walk(source_dir):
         for filename in filenames:
             if filename.rpartition(".")[-1] in (
                 "py",
                 "pyi",
             ) and not filename.startswith("utils.py"):
                 filepaths.append(os.path.join(root, filename))
 
-    unasync.unasync_files(filepaths, rules)
+    unasync.unasync_files(filepaths, [rule])
+
+    if cleanup_patterns:
+        cleanup(source_dir, output_dir, cleanup_patterns)
+
+    if format:
+        format_dir(source_dir)
+        format_dir(output_dir)
+
+
+def main():
+    run(
+        rule=unasync.Rule(
+            fromdir="/elasticsearch/_async/client/",
+            todir="/elasticsearch/_sync/client/",
+            additional_replacements={
+                # We want to rewrite to 'Transport' instead of 'SyncTransport', etc
+                "AsyncTransport": "Transport",
+                "AsyncElasticsearch": "Elasticsearch",
+                # We don't want to rewrite this class
+                "AsyncSearchClient": "AsyncSearchClient",
+                # Handling typing.Awaitable[...] isn't done yet by unasync.
+                "_TYPE_ASYNC_SNIFF_CALLBACK": "_TYPE_SYNC_SNIFF_CALLBACK",
+            },
+        ),
+    )
+
+    run(
+        rule=unasync.Rule(
+            fromdir="elasticsearch/helpers/vectorstore/_async/",
+            todir="elasticsearch/helpers/vectorstore/_sync/",
+            additional_replacements={
+                "AsyncBM25": "BM25",
+                "AsyncDenseVector": "DenseVector",
+                "AsyncDenseVectorScriptScore": "DenseVectorScriptScore",
+                "AsyncElasticsearch": "Elasticsearch",
+                "AsyncElasticsearchEmbeddings": "ElasticsearchEmbeddings",
+                "AsyncEmbeddingService": "EmbeddingService",
+                "AsyncRetrievalStrategy": "RetrievalStrategy",
+                "AsyncSparseVector": "SparseVector",
+                "AsyncTransport": "Transport",
+                "AsyncVectorStore": "VectorStore",
+                "async_bulk": "bulk",
+                "_async": "_sync",
+            },
+        ),
+        cleanup_patterns=[
+            "/^import asyncio$/d",
+            "/^import pytest_asyncio*/d",
+            "/ *@pytest.mark.asyncio$/d",
```

Comment:
These replacements would be necessary only for tests and examples, so they would not be needed on the vectorstore directory.